### PR TITLE
Feature: Update FBBG to work on Web and Desktop

### PR DESCRIPTION
**Pull request by csells**
suggested fix for https://github.com/ajinasokan/flutter_fgbg/issues/16

### CODE REVIEW

**Comment by csells** [comment]
I'm since tested this on my own app that needs to run on both Android and the Mac and it works great. On Android, it works better than the default AppLifecycleListener-based implementation for reasons you already know and on the Mac it falls back to the default. win-win.

File: ios/Classes/FlutterFgbgPlugin.swift
Comment (ajinasokan) [inline ios/Classes/FlutterFgbgPlugin.swift:1]: Was this file automatically added? This seems like the sample code from the plugin template. Actual plugin code is in [SwiftFlutterFgbgPlugin.swift](https://github.com/ajinasokan/flutter_fgbg/blob/master/ios/Classes/SwiftFlutterFgbgPlugin.swift)

File: lib/flutter_fgbg.dart [hunk 1]
```diff
@@ -49,18 +54,47 @@ class FGBGNotifier extends StatefulWidget {
...
   void initState() {
     super.initState();
-    subscription = FGBGEvents.stream.listen((event) {
-      widget.onEvent.call(event);
-    });
+
+    // NOTE: short-circuit the platform check on web to avoid a crash
+    if (!kIsWeb && (Platform.isAndroid || Platform.isIOS)) {
+      // use mobile-specific implementations
+      print('subscribing:');
```
Comment (ajinasokan) [inline lib/flutter_fgbg.dart:66]: These print statements can be removed

File: lib/flutter_fgbg.dart [hunk 2]
```diff
@@ -49,18 +54,47 @@ class FGBGNotifier extends StatefulWidget {
...
+      subscription = FGBGEvents.instance.stream.listen((event) {
+        widget.onEvent.call(event);
+      });
+    } else {
+      // use Flutter's default implementation
+      listener = AppLifecycleListener(
+        onStateChange: (state) {
+          // check if we're ignoring events right now
+          if (FGBGEvents._ignoreEvent) {
+            print('ignoring events');
```
Comment (ajinasokan) [inline lib/flutter_fgbg.dart:76]: This one too

**Review by ajinasokan — COMMENTED** [review]
Everything else looks good! I will merge, test and push a release after these. Or if you are occupied, let me know, I can fix these.

**Comment by csells** [comment]
I'm heads down today. If you wouldn't mind making those fixes and pushing an updated release, that would be appreciated.

File: ios/Classes/FlutterFgbgPlugin.swift
Comment (csells) [reply ios/Classes/FlutterFgbgPlugin.swift:1]: I didn't add that manually; it must've been automatic.

**Comment by ajinasokan** [comment]
No problem. I can work on it. 👍🏽

**Pull request merged**